app/both/collections/residents.js

```diff
@@ -73,98 +72,120 @@ Residents.helpers({
     // Get all activities involving resident
     // make sure activities are in the past (i.e. not planned)
     //  sort in reverse order by activity date
-    return Activities.find({'residentIds': residentId, activityDate: {$lte: now}}, {sort : {activityDate:  -1} });
+    return Activities.find(
+      { residentIds: residentId, activityDate: { $lte: now } },
+      { sort: { activityDate: -1 } }
+    );
   },
-  'recentActivities': function () {
+  recentActivities: function() {
     // Get resident ID
     var residentId = this._id;
 
     // Date two weeks ago
-    var twoWeeksAgo = moment().subtract(2, "weeks").toDate();
+    var twoWeeksAgo = moment()
+      .subtract(2, 'weeks')
+      .toDate();
 
     // Date today
     var now = new Date();
 
     // Get all activities involving resident
     // make sure activities are in the past (i.e. not planned)
     //  sort in reverse order by activity date
-    return Activities.find({'residentIds': residentId, activityDate: {$gte: twoWeeksAgo, $lte: now}}, {sort : {activityDate:  -1} });
-  }
+    return Activities.find(
+      {
+        residentIds: residentId,
+        activityDate: { $gte: twoWeeksAgo, $lte: now },
+      },
+      { sort: { activityDate: -1 } }
+    );
+  },
 });
 
 Residents.allow({
-  insert: function (userId, doc) {
-    const schemaType = "resident";
-    const action = "insert";
-    
+  insert: function(userId, doc) {
+    const schemaType = 'resident';
+    const action = 'insert';
+
     return checkUserPermissions({ schemaType, action, userId, doc });
   },
-  update: function (userId, doc) {
-    const schemaType = "resident";
-    const action = "update";
+  update: function(userId, doc) {
+    const schemaType = 'resident';
+    const action = 'update';
     const residentId = doc._id;
-    const activeResidency = Residencies.findOne({ $and: [{ residentId }, { moveOut: { $exists: false } }] });
-    doc.homeId = activeResidency.homeId
-    
+    const activeResidency = Residencies.findOne({
+      $and: [{ residentId }, { moveOut: { $exists: false } }],
+    });
+    doc.homeId = '';
+    if (activeResidency) {
+      doc.homeId = activeResidency.homeId;
+    }
+
     return checkUserPermissions({
       schemaType,
       action,
       userId,
-      doc
+      doc,
     });
   },
-  remove: function (userId, doc) {
-    const schemaType = "resident";
-    const action = "remove";
+  remove: function(userId, doc) {
+    const schemaType = 'resident';
+    const action = 'remove';
     const residentId = doc._id;
-    const activeResidency = Residencies.findOne({ $and: [{ residentId }, { moveOut: { $exists: false } }] });
-    doc.homeId = activeResidency.homeId
+    const activeResidency = Residencies.findOne({
```

Comment:
Give this code some breathing room with line breaks and comments for the dense lines.